### DISCUSSION TITLE
The .get<>(); method is not converting the values

### DISCUSSION THREAD

**Discussion by luizfernandonb** [nlohmann/json#2527] [Q&A] [answered]
I'm trying to get a float value in Json and the `.get<>();` method is not working

#### What is the issue you have?

[This](https://economia.awesomeapi.com.br/json/all/USD-BRL) is Json, and I'm talking about the `high` property which is a `float/double`

#### Please describe the steps to reproduce the issue.

Copy this code and run it on a Visual Studio project that is including the `nlohmann/json.hpp` library and the `cpr/cpr.h` library

```cpp
cpr::Response response = cpr::Get(cpr::Url { "https://economia.awesomeapi.com.br/json/all/USD-BRL" });

nlohmann::json json_response = nlohmann::json::parse(response.text);

nlohmann::json high = json_response["USD"]["high"];
float high_value = high.get<float>();
```

#### What is the expected behavior?

The last line of code to be json's `high` property

#### And what is the actual behavior instead?

An exception

![image](https://user-images.githubusercontent.com/42288418/101982143-a3515a00-3c50-11eb-92cc-f00d8337628f.png)

Translation:

> Untreated exception on 0x76C9A892 in ToastDollar.exe: Microsoft C++ exception: nlohmann::detail::type_error at memory location 0x00CFF524.

#### Which compiler and operating system are you using?

- Compiler: `The Visual Studio Compiler (MSVC)`
- Operating system: `Windows 10 2004`

#### Which version of the library did you use?

- [x] latest release version 3.9.1
- [ ] other release - please state the version: ___
- [ ] the `develop` branch

**Comment by nlohmann** [comment, marked as answer]
This is not a bug. The value is a string and get() does not perform string-to-float conversion. The text of the exception describes that type mismatch.

**Reply by luizfernandonb** [reply]
> This is not a bug. The value is a string and get() does not perform string-to-float conversion. The text of the exception describes that type mismatch.

But in the documentation you're saying it's possible, look at the last line

```cpp

    // create a JSON value with different types
    json json_types =
    {
        {"boolean", true},
        {
            "number", {
                {"integer", 42},
                {"floating-point", 17.23}
            }
        },
        {"string", "Hello, world!"},
        {"array", {1, 2, 3, 4, 5}},
        {"null", nullptr}
    };

    // use explicit conversions
    auto v1 = json_types["boolean"].get<bool>();
    auto v2 = json_types["number"]["integer"].get<int>();
    auto v3 = json_types["number"]["integer"].get<short>();
    > auto v4 = json_types["number"]["floating-point"].get<float>();
```

https://nlohmann.github.io/json/api/basic_json/get/

**Reply by nlohmann** [reply]
In these cases, numeric values are converted:

- v2 converts a JSON number to an `int`.
- v3 converts a JSON number to a `short`.
- v4 converts a JSON number to a `float`.

In your example, you want to convert a JSON string `"5.0883"` to a `float` which is not supported.

**Reply by luizfernandonb** [reply]
> In these cases, numeric values are converted:
> 
> * v2 converts a JSON number to an `int`.
> * v3 converts a JSON number to a `short`.
> * v4 converts a JSON number to a `float`.
> 
> In your example, you want to convert a JSON string `"5.0883"` to a `float` which is not supported.

How is it not supported? I don't understand, there's nothing wrong with this number.

**Reply by nlohmann** [reply]
The value is a string `"5.0883"` and not a number `5.0883`.

**Reply by luizfernandonb** [reply]
> The value is a string `"5.0883"` and not a number `5.0883`.

Now I understood, thanks, I thought it was possible to convert directly from string to float, so I will have to use the method of C++ itself

But would it be possible in the future to implement something to convert directly from string to float?
